code/datums/keybindings/emote.dm

```diff
@@ -0,0 +1,594 @@
+/datum/keybinding/emote/glare
+	linked_emote = /datum/emote/living/glare
+	name = "Glare"
+
+/datum/keybinding/emote/grin
+	linked_emote = /datum/emote/living/grin
+	name = "Grin"
+
+/datum/keybinding/emote/grimace
+	linked_emote = /datum/emote/living/grimace
+	name = "Grimace"
+
+/datum/keybinding/emote/groan
+	linked_emote = /datum/emote/living/groan
+	name = "Groan"
+
+/datum/keybinding/emote/look
+	linked_emote = /datum/emote/living/look
+	name = "Look"
+
+/datum/keybinding/emote/bshake
+	linked_emote = /datum/emote/living/bshake
+	name = "Shake"
+
+/datum/keybinding/emote/shudder
+	linked_emote = /datum/emote/living/shudder
+	name = "Shudder"
+
+/datum/keybinding/emote/point
+	linked_emote = /datum/emote/living/point
+	name = "Point"
+
+/datum/keybinding/emote/pout
+	linked_emote = /datum/emote/living/pout
+	name = "Pout"
+
+/datum/keybinding/emote/scream
+	linked_emote = /datum/emote/living/scream
+	name = "Scream"
+
+/datum/keybinding/emote/shake
+	linked_emote = /datum/emote/living/shake
+	name = "Head Shake"
+
+/datum/keybinding/emote/shiver
+	linked_emote = /datum/emote/living/shiver
+	name = "Shiver"
+
+/datum/keybinding/emote/sigh
+	linked_emote = /datum/emote/living/sigh
+	name = "Sigh"
+
+/datum/keybinding/emote/happy
+	linked_emote = /datum/emote/living/sigh/happy
+	name = "Sigh (Happy)"
+
+/datum/keybinding/emote/sit
+	linked_emote = /datum/emote/living/sit
+	name = "Sit"
+
+/datum/keybinding/emote/smile
+	linked_emote = /datum/emote/living/smile
+	name = "Smile"
+
+/datum/keybinding/emote/smug
+	linked_emote = /datum/emote/living/smug
+	name = "Smug"
+
+/datum/keybinding/emote/sniff
+	linked_emote = /datum/emote/living/sniff
+	name = "Sniff"
+
+/datum/keybinding/emote/snore
+	linked_emote = /datum/emote/living/snore
+	name = "Snore"
+
+/datum/keybinding/emote/stare
+	linked_emote = /datum/emote/living/stare
+	name = "Stare"
+
+/datum/keybinding/emote/stretch
+	linked_emote = /datum/emote/living/strech
+	name = "Stretch"
+
+/datum/keybinding/emote/sulk
+	linked_emote = /datum/emote/living/sulk
+	name = "Sulk"
+
+/datum/keybinding/emote/sway
+	linked_emote = /datum/emote/living/sway
+	name = "Sway"
+
+/datum/keybinding/emote/swear
+	linked_emote = /datum/emote/living/swear
+	name = "Swear"
+
+/datum/keybinding/emote/tilt
+	linked_emote = /datum/emote/living/tilt
+	name = "Tilt"
+
+/datum/keybinding/emote/tremble
+	linked_emote = /datum/emote/living/tremble
+	name = "Tremble"
+
+/datum/keybinding/emote/twitch
+	linked_emote = /datum/emote/living/twitch
+	name = "Twitch (Violent)"
+
+/datum/keybinding/emote/twitch_s
+	linked_emote = /datum/emote/living/twitch_s
+	name = "Twitch"
+
+/datum/keybinding/emote/whimper
+	linked_emote = /datum/emote/living/whimper
+	name = "Whimper"
+
+/datum/keybinding/emote/wsmile
+	linked_emote = /datum/emote/living/wsmile
+	name = "Smile (Weak)"
+
+/datum/keybinding/emote/carbon
+	category = KB_CATEGORY_EMOTE_CARBON
+
+/datum/keybinding/emote/carbon/can_use(client/C, mob/M)
+	return iscarbon(M) && ..()
```

Comment:
This is fine, but again, probably unnecessary since the emote shouldn't work if the mob isn't a carbon mob.

Reply:
These also help reduce the amount of keys we need to check during ssinput looping, so it's worth keeping